src/jailer/src/main.rs

```diff
@@ -328,6 +328,33 @@ fn sanitize_process() {
     }
 }
 
+#[cfg(target_arch = "aarch64")]
+// Safe because this syscall takes static parameters and
+// we check the output.
+unsafe fn enable_ssbd_mitigation() {
+    // Parameters for `prctl`
+    // TODO: generate bindings for these from the kernel sources.
+    // https://elixir.bootlin.com/linux/v4.17/source/include/uapi/linux/prctl.h#L212
+    const PR_SET_SPECULATION_CTRL: i32 = 53;
+    const PR_SPEC_STORE_BYPASS: u64 = 0;
+    const PR_SPEC_FORCE_DISABLE: u64 = 1u64 << 3;
+
+    let ret = libc::prctl(
+        PR_SET_SPECULATION_CTRL,
+        PR_SPEC_STORE_BYPASS,
+        PR_SPEC_FORCE_DISABLE,
+        0,
+        0,
+    );
+
+    if ret < 0 {
+        println!("Could not enable SSBD mitiagtion through the prctl interface.");
```

Comment:
I'd print the last os error here and in the log line below. It can be useful for debugging

Reply:
Looking at the [`prctl` man page](https://man7.org/linux/man-pages/man2/prctl.2.html) and the[ kernel documentation](https://www.kernel.org/doc/html/latest/userspace-api/spec_ctrl.html), the only fail we are interested in is `EINVAL`, which means the architecture/host kernel do not support this mitigation.

I can add the error code after the message, but I don't personally see the added value.

Reply:
Them we can make the error message sound like "SSBD mitigation not available on this architecture/kernel".

Reply:
The problem is that SSBD mitigation might be available on that architecture/kernel, just not through the `prctl` interface but through the kernel command line parameter `ssbd=force-on`. The `prctl` interface supports SSBD mitigation since Linux 4.17 and it has also been ported to Amazon Linux 4.14 variants.

Reply:
I think we can also get a `ENXIO` - Control of the selected speculation misfeature is not possible, not sure.

I'd really like to default to also printing the OS error. You never know when it might come handy when looking through the logs

Reply:
`ENXIO` is not possible because we only run this on `aarch64`, and the setting mentioned in the man page which could result in `ENXIO` is `x86` only.

Regardless, I appended the error code to the message.